services/src/main/java/io/scalecube/services/ServiceProxyFactory.java

```diff
@@ -0,0 +1,118 @@
+package io.scalecube.services;
+
+import io.scalecube.services.annotations.ServiceProcessor;
+import io.scalecube.services.routing.Router;
+import io.scalecube.services.routing.RouterFactory;
+import io.scalecube.transport.Message;
+
+import com.google.common.reflect.Reflection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ServiceProxyFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServiceProxyFactory.class);
+
+  /**
+   * used to complete the request future with timeout exception in case no response comes from service.
+   */
+  private static final ScheduledExecutorService delayer = Executors.newScheduledThreadPool(1);
```

Comment:
I don't like that we create so much ScheduledExecutor's, we should share resource as much as possible,
Also we should correctly shutdown thread pools or at least make them daemon